app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,41 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!, only: [:index, :create]
+  before_action :set_item, only: [:index, :create]
+
+  def index
+    if user_signed_in? && current_user.id != @item.user_id
```

Comment:
```ruby
if user_signed_in? && current_user.id != @item.user_id
```
こちらですが、Trelloの下記の要件を満たすために現在の`ユーザーがログインしている、且つ、出品者とログインユーザーが一致しない`という条件に加えて**あるいは、商品が売り切れていない**という条件を追加してあげましょう。
```
- ログイン・ログアウトの状態に関わらず、URLを直接入力して売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移すること
```

Reply:
ありがとうございます！上記の条件を追加しました。また、ログインページに遷移してしまうため以下のコードからindexを削除してroot_pathでトップページに遷移するように修正致しました。
```suggestion
    before_action :authenticate_user!, only: [:index, :create]
```